lib/seahorn/BvOpSem2FatMemMgr.cc

```diff
@@ -193,20 +239,28 @@ class FatMemManager : public OpSemMemManager {
   /// \return symbolic value of the read integer
   Expr loadIntFromMem(FatPtrTy ptr, FatMemValTy mem, unsigned byteSz,
                       uint64_t align) override {
-    return m_mem.loadIntFromMem(mkRawPtr(ptr), mkRawMem(mem), byteSz, align);
+    return m_main.loadIntFromMem(mkRawPtr(ptr), mkRawMem(mem), byteSz, align);
   }
 
   /// \brief Loads a pointer stored in memory
   /// \sa loadIntFromMem
   FatPtrTy loadPtrFromMem(FatPtrTy ptr, FatMemValTy mem, unsigned byteSz,
                           uint64_t align) override {
-    return mkFatPtr(
-        m_mem.loadPtrFromMem(mkRawPtr(ptr), mkRawMem(mem), byteSz, align));
+    MemValTy rawVal =
+        m_main.loadPtrFromMem(mkRawPtr(ptr), mkRawMem(mem), byteSz, align);
+    MemValTy slot0Val =
+        m_slot0.loadIntFromMem(mkRawPtr(ptr), mkSlot0Mem(mem), byteSz, align);
+    MemValTy slot1Val =
+        m_slot1.loadIntFromMem(mkRawPtr(ptr), mkSlot1Mem(mem), byteSz, align);
+    LOG("opsem", WARN << "ptr:" << rawVal << " slot0:" << slot0Val
+                      << " slot1:" << slot1Val << "\n");
+    LOG("opsem", WARN << "FatMem:" << mem);
+    return mkFatPtr(rawVal, slot0Val, slot1Val);
   }
 
   /// \brief Pointer addition with numeric offset
   FatPtrTy ptrAdd(FatPtrTy ptr, int32_t _offset) const {
-    PtrTy rawPtr = m_mem.ptrAdd(mkRawPtr(ptr), _offset);
+    PtrTy rawPtr = m_main.ptrAdd(mkRawPtr(ptr), _offset);
```

Comment:
This is an important difference from what Xiang will usually generate. We are changing gep to copy fat while in his case a gep will follow by a copy of fat. Not yet convinced what is the best course of action. But this does not affect current tests. Just something to note in code.

Reply:
noted in code.